### DISCUSSION TITLE
Inconsistent Behavior Switch to Dark Theme

### DISCUSSION THREAD

**Discussion by suiboli314** [tailwindlabs/tailwindcss#12721] [Help] [answered]
There appears to be a potential inconsistency in the dark theme's behavior for an element structured as follows:

```
- div (dark:drop-shadow-none)
|- div (position:fixed)
```

To better illustrate and address this concern, I have built a [demo](https://codesandbox.io/p/devbox/dark-theme-inconsistency-t4wzxs?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clr1d758s0007356g2weu0l9q%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clr1d758s0002356g5041my62%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clr1d758s0004356g6fmsq4v8%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clr1d758s0006356gcb4bxe71%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B51.4305093106425%252C48.5694906893575%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clr1d758s0002356g5041my62%2522%253A%257B%2522id%2522%253A%2522clr1d758s0002356g5041my62%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clr1dxmcs0002356g8wh3l4ze%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A19%252C%2522startColumn%2522%253A15%252C%2522endLineNumber%2522%253A19%252C%2522endColumn%2522%253A15%257D%255D%252C%2522filepath%2522%253A%2522%252Fapp%252Fpage.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clr1dxmcs0002356g8wh3l4ze%2522%257D%252C%2522clr1d758s0006356gcb4bxe71%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clr1d758s0005356gu85ntmdi%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clr1d758s0006356gcb4bxe71%2522%252C%2522activeTabId%2522%253A%2522clr1d758s0005356gu85ntmdi%2522%257D%252C%2522clr1d758s0004356g6fmsq4v8%2522%253A%257B%2522id%2522%253A%2522clr1d758s0004356g6fmsq4v8%2522%252C%2522activeTabId%2522%253A%2522clr1f1b1t00wv356gst1jql24%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522clr1f1b1t00wv356gst1jql24%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A10%257D).

**Comment by fordaaron** [comment, marked as answer]
This is expected behavior of the [filter](https://arc.net/l/quote/jugyrayr) spec. Any children of an element with a filter applied, such as drop shadow, results in the creation of a containing block for absolute and fixed positioned descendants.

This means that the descendent `<div class="fixed ...">` will be fixed relative to `<div class="sticky dark:drop-shadow-none ...">` rather that the root of the document, which is why it is moving to the top of the screen.
